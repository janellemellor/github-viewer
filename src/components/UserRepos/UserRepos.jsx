import React from 'react';
import PropTypes from 'prop-types';
import UserRepo from '../UserRepo/UserRepo.jsx';

const UserRepos = ({ repos }) => {
  const allRepos = repos.map(repo => <UserRepo key ={repo.id} {...repo} />);
  return (
    <ul>
      {allRepos}
    </ul>
  );
};

UserRepos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired, 
    html_url: PropTypes.string.isRequired   
  })).isRequired
};

export default UserRepos;
