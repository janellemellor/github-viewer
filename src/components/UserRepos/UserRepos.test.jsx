import React from 'react';
import { shallow } from 'enzyme';
import UserRepos from './UserRepos.jsx';

describe('UserRepos', () => {
  it('matches a snapshot', () => {
    const repos = [
      {
        id: 1, 
        name: 'backend',
        html_url: 'https://github.com/mental-health-twitterbot/backend'
      },
      {
        id: 2, 
        name: 'mongo-aggregation',
        html_url: 'https://github.com/janellemellor/mongo-aggregation'
      }
    ];    
    const wrapper = shallow(<UserRepos repos={repos} />);
    expect(wrapper).toMatchSnapshot();
  });
});

