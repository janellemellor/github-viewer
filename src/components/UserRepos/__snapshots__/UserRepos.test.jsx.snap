// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepos matches a snapshot 1`] = `
<ul>
  <UserRepo
    html_url="https://github.com/mental-health-twitterbot/backend"
    id={1}
    key="1"
    name="backend"
  />
  <UserRepo
    html_url="https://github.com/janellemellor/mongo-aggregation"
    id={2}
    key="2"
    name="mongo-aggregation"
  />
</ul>
`;
